engine/src/main/java/org/terasology/rendering/dag/AbstractNode.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import com.google.common.collect.Sets;
+import java.util.Set;
+
+/**
+ * TODO: Add javadocs
+ */
+public abstract class AbstractNode implements Node {
+    private Set<StateChange> stateChanges;
+
+    protected AbstractNode() {
+        stateChanges = Sets.newLinkedHashSet();
+    }
+
+    protected boolean addStateChange(StateChange stateChange) {
+        return stateChanges.add(stateChange);
+    }
+
+    @Override
+    public Set<StateChange> getStateChanges() {
+        return stateChanges;
+    }
+}
```

Comment:
Scrap this comment. I wrote it before I noticed your fifth commit in which you do exactly what I'm suggesting.